restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/util/RCUtils.java

```diff
@@ -173,4 +177,90 @@ public static void validateConnectionParms(HashMap<String, Object> parameters) t
+              TextUtils.isEmpty((String) parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ACCOUNT_EMAIL))) {
+         throw new RCException(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_ACCOUNT_EMAIL_MISSING);
+      }
+
+      if (!parameters.containsKey(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ACCOUNT_PASSWORD) ||
+              parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ACCOUNT_PASSWORD) == null ||
+              TextUtils.isEmpty((String) parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ACCOUNT_PASSWORD))) {
+         throw new RCException(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_ACCOUNT_PASSWORD_MISSING);
+      }
+
+      if (!parameters.containsKey(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_PUSH_DOMAIN) ||
+              parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_PUSH_DOMAIN) == null ||
+              TextUtils.isEmpty((String) parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_PUSH_DOMAIN))) {
+         throw new RCException(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_PUSH_DOMAIN_MISSING);
+      }
+
+      if (!parameters.containsKey(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_HTTP_DOMAIN) ||
+              parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_HTTP_DOMAIN) == null ||
+              TextUtils.isEmpty((String) parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_HTTP_DOMAIN))) {
+         throw new RCException(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_RESTCOMM_DOMAIN_MISSING);
+      }
+
+      //when binding is missing we need to register for push
+      if (storageManagerInterface.getString(FcmConfigurationHandler.FCM_BINDING, null) == null){
+         return true;
+      }
+
+      if (!parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_FCM_SERVER_KEY).equals
+              (storageManagerInterface.getString(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_FCM_SERVER_KEY, null))) {
+         return true;
+      }
+
+      if (!parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_APPLICATION_NAME).equals
+              (storageManagerInterface.getString(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_APPLICATION_NAME, null))) {
+         return true;
+      }
+
+      if (!parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ACCOUNT_PASSWORD).equals
+              (storageManagerInterface.getString(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ACCOUNT_PASSWORD, null))) {
+         return true;
+      }
+
+      if (!parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_PUSH_DOMAIN).equals
+              (storageManagerInterface.getString(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_PUSH_DOMAIN, null))) {
+         return true;
+      }
+
+      if (!parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_HTTP_DOMAIN).equals
+              (storageManagerInterface.getString(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_HTTP_DOMAIN, null))) {
+         return true;
+      }
+
+      return false;
```

Comment:
Let's add a log saying whether shouldRegisterForPush() returned true or false; I think it would be handy for troubleshooting